build_all/make_sdk.py

```diff
@@ -0,0 +1,192 @@
+    for filePath in fileList:
+        try:
+            os.remove(filePath)
+        except:
+            print("Error while deleting file : ", filePath)
+
+
+def pattern_delete_folder(pattern): # helper method
+    # Get a list of all the file paths that match the pattern in specified directory
+    fileList = glob.glob(pattern)
+    # Iterate over the list of filepaths & remove each file.
+    for filePath in fileList:
+        try:
+            shutil.rmtree(filePath)
+        except:
+            print("Error while deleting filepath : ", filePath)
+
+def line_prepender(filename, line):
+    with open(filename, 'r+') as f:
+        content = f.read()
+        f.seek(0, 0)
+        f.write(line.rstrip('\r\n') + '\n' + content)
+
+def line_appender(filename, line):
+    with open(filename, 'a') as f:
+        f.write(line.rstrip('\r\n') + '\n')
+
+def usage():
+    # Iterates through command dictionary to print out script's opt usage
+    usage_txt = "make_sdk.py accepts the following arguments: \r\n"
+
+    for commands in commands_dict.cmds:
+        usage_txt += " - %s: " %commands + commands_dict.cmds[commands]['text'] + "\r\n"
+
+    return usage_txt
+
+def parse_opts():
+    options, remainder = getopt.gnu_getopt(sys.argv[1:], 'ho:d:', ['output', 'help', 'device'])
+    # print('OPTIONS   :', options)
+    for opt, arg in options:
+        if opt in ('-h', '--help'):
+            print(usage())
+        elif opt in ('-o', '--output'):
+            commands_dict.output_path = arg
+            
+
+def run():
+    # set up paths for copying
+    output_path = os.path.abspath(commands_dict.output_path)
+    arduino_repo_root = os.path.abspath('../')
+    print(output_path)
+    arduino_pal_path = arduino_repo_root+'/pal/'
+    azure_iot_sdk_path = arduino_repo_root+'/sdk/'
+    AzureIoTHub_path = output_path+'/AzureIoTHub/'
+    AzureIoTProtocolHTTP_path = output_path+'/AzureIoTProtocol_HTTP/'
+    AzureIoTProtocolMQTT_path = output_path+'/AzureIoTProtocol_MQTT/'
+    AzureIoTUtility_path = output_path+'/AzureIoTUtility/'
+    AzureIoTSocketWiFi_path = output_path+'/AzureIoTSocket_WiFi/'
+    # AzureIoTSocketEthernet_path = output_path+'/AzureIoTSocket_Ethernet/'
+    AzureUHTTP_path = AzureIoTProtocolHTTP_path+'src/azure_uhttp_c/'
+    AzureUMQTT_path = AzureIoTProtocolMQTT_path+'src/azure_umqtt_c/'
+    SharedUtility_path = AzureIoTUtility_path+'src/azure_c_shared_utility/'
+    Adapters_path = AzureIoTUtility_path+'src/adapters/'
+    Macro_Utils_path = AzureIoTUtility_path+'src/azure_c_shared_utility/azure_macro_utils/'
+    Hub_Macro_Utils_path = AzureIoTHub_path+'src/azure_macro_utils/'
+    Umock_c_path = AzureIoTUtility_path+'src/umock_c/'
+    sdk_path = AzureIoTHub_path+'src/'
+    internal_path = AzureIoTHub_path+'src/internal/'
+
+    if (os.path.exists(output_path)):
+        #clear it out
+        pattern_delete_folder(output_path+'/Azure*')
+    else:
+        os.mkdir(output_path)
```

Comment:
Can this check be done first thing in the run?

Reply:
It's not really a check so much as it is resetting the folder structure.